internal/util/certs.go

```diff
@@ -61,3 +78,61 @@ func LoadCerts() (*x509.CertPool, error) {
 
 	return ngrokCertPool, loadCertsOnceErr
 }
+
+func watchCertsDirWithHandler(path string, onChange func()) error {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return err
+	}
+	defer watcher.Close()
+
+	if err := watcher.Add(path); err != nil {
+		return err
+	}
+
+	for {
+		select {
+		case event, ok := <-watcher.Events:
+			if !ok {
+				return nil
+			}
+			if event.Op&(fsnotify.Write|fsnotify.Create|fsnotify.Remove|fsnotify.Rename) != 0 {
+				onChange()
+				return nil
+			}
+		case err, ok := <-watcher.Errors:
+			if !ok {
+				return nil
+			}
+			return err
+		}
+	}
+}
+
+// watchCertsDir watches for changes in the certs directory and exits on any update
+func watchCertsDir() {
+	_ = watchCertsDirWithHandler(customCertsPath, func() {
+		if shouldExitOnCertChange() {
+			ctrl.Log.Info("Detected changes in custom certs directory, reloading certs")
+			os.Exit(0)
+		} else {
+			ctrl.Log.Info("Detected changes in custom certs directory, but NGROK_OPERATOR_EXIT_ON_CERT_CHANGE is not set to true, so not exiting")
```

Comment:
Nit: I think Exit is how the implementation detail here works, but as a user "Exit" would concern me vs something like "Restart". we just happen to know here that its the process exiting which k8s will restart.